app/server/endpoint/src/main/java/io/syndesis/server/endpoint/v1/handler/integration/IntegrationHandler.java

```diff
@@ -98,10 +99,19 @@ public Integration create(@Context final SecurityContext sec, final Integration
         final Integration updatedIntegration = new Integration.Builder().createFrom(encryptedIntegration)
             .createdAt(System.currentTimeMillis()).build();
 
+        validateLabels(updatedIntegration);
```

Comment:
Could we use JSR 303 validation instead here? I don't think throwing `IllegalArgumentException` is what we do right now, we have a protocol in place for validation errors and how they're signaled to the UI via `ConstraintViolationExceptionMapper` et al. Look at `@UniqueProperty` and [configuration](https://github.com/syndesisio/syndesis/tree/1.13.x/app/server/dao/src/main/resources/META-INF/validation).